recipes/pytest-executable/meta.yaml

```diff
@@ -0,0 +1,46 @@
+{% set name = "pytest_executable" %}
+{% set version = "0.4.0" %}
+
+package:
+  name: pytest-executable
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 316b7982bcdb923ad6b8453f54b348bf81328dcdcecad212034acdf592bd3a4d
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - python >=3.6
+    - pytest >=4.3,<6
+    - pyyaml >=3,<6
+    - jsonschema >=2,<4
+    - jinja2 >=2.7,<3
+    - dataclasses
+
+test:
+  commands:
+    - "pytest --traceconfig | grep pytest-executable-{{ version }}"
```

Comment:
a minor nit:

while conda-forge is only going to build this on linux, the theory of `noarch` is that it _could_ build on the lowest common denominator (e.g. vanilla windows).

It is _possible_ to do this as written (but wouldn't the dots get expanded?), by adding the `- posix #[win]` as a test dependency. Likely you can use github search to find recipes that contain `noarch` and `grep`. i, personally, would move this to `run_test.py` which will automatically be run by conda on any compatible platform (which is, well, all of them) and won't bring in any extra dependencies.

Another command I find worth adding when there are non-trivial dependency pins, is:

```bash
test:
  requires:
    - pip
  commands:
    - python -m pip check
```

Reply:
Thanks, I've added the pip check.